azurerm/resource_arm_virtual_network_gateway_connection.go

```diff
@@ -369,6 +379,10 @@ func resourceArmVirtualNetworkGatewayConnectionRead(d *schema.ResourceData, meta
 		d.Set("shared_key", conn.SharedKey)
 	}
 
+	if string(conn.ConnectionProtocol) != "" {
+		d.Set("connection_protocol", string(conn.ConnectionProtocol))
+	}
```

Comment:
can we remove this `if` statement here?

Reply:
I was thinking about that, but that API documentation does want to have either IKEv1 or IKEv2 as a value. Also wouldn't that be a breaking change? Currently the value is never set. If we suddenly start to set it, it would cause all connections to be recreated, as the type can't be changed without recreation.

Reply:
It might be best to mark it as computed in that case?